scripts/cleanup_e2e_resources.ts

```diff
@@ -342,3 +360,77 @@ for (const staleBranch of allStaleBranches) {
     );
   }
 }
+
+const listAllStaleRoles = async (): Promise<Array<Role>> => {
+  let nextToken: string | undefined = undefined;
+  const roles: Array<Role> = [];
+  do {
+    const listRolesCommandOutput: ListRolesCommandOutput = await iamClient.send(
+      new ListRolesCommand({
+        Marker: nextToken,
+      })
+    );
+    nextToken = listRolesCommandOutput.Marker;
+    if (listRolesCommandOutput.Roles) {
+      listRolesCommandOutput.Roles.filter(
+        (role: Role) =>
+          (role.RoleName?.startsWith(TEST_AMPLIFY_RESOURCE_PREFIX) ||
+            role.RoleName?.startsWith(TEST_CDK_RESOURCE_PREFIX)) &&
```

Comment:
Are roles the only test resources that can have `TEST_CDK_RESOURCE_PREFIX`? Wondering if this should be extended to other resources as well.

Reply:
yes, these come from auth stacks only so far.

Reply:
We don't orphan other resources from cdk tests so far, so we'll expand this later as needed.